core/src/main/scala/kafka/tools/ConsoleProducer.scala

```diff
@@ -251,12 +255,22 @@ object ConsoleProducer {
     val options = parser.parse(args : _*)
     if(args.length == 0)
       CommandLineUtils.printUsageAndDie(parser, "Read data from standard input and publish it to Kafka.")
-    CommandLineUtils.checkRequiredArgs(parser, options, topicOpt, brokerListOpt)
 
     val useOldProducer = options.has(useOldProducerOpt)
-    val topic = options.valueOf(topicOpt)
+    val bootstrapServer = options.valueOf(bootstrapServerOpt)
```

Comment:
The general pattern over all files should be that `bootstrapServer` has the value of `bootstrapServerOpt` _or_ `brokerListOpt`. Since we validate that only one of them can be set, this is OK. If we do it that way, we don't have to check two fields everywhere (or worse forget to check one of them like it seems to be happening here in `getNewProducerProps`.